packages/runtime/container-runtime/src/opTelemetry.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDeltaManager } from "@fluidframework/container-definitions";
+import {
+    IDocumentMessage,
+    ISequencedDocumentMessage,
+} from "@fluidframework/protocol-definitions";
+import { isSystemMessage } from "@fluidframework/protocol-base";
+
+export class OpTracker {
+    /**
+     * Used for storing the message content size when
+     * the message is pushed onto the inbound queue.
+     */
+    private readonly messageSize = new Map<number, number>();
+    private _nonSystemOpCount: number = 0;
+    public get nonSystemOpCount(): number {
+        return this._nonSystemOpCount;
+    }
+
+    private _opsSizeAccumulator: number = 0;
+    public get opsSizeAccumulator(): number {
+        return this._opsSizeAccumulator;
+    }
+
+    public constructor(
+        deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
+        disabled: boolean,
+    ) {
+        if (disabled) {
+            return;
+        }
+
+        // Record the message content size when we receive it.
+        // We should not log this value, as summarization can happen between the time the message
+        // is received and until it is processed (the 'op' event).
+        deltaManager.inbound.on("push", (message: ISequencedDocumentMessage) => {
+            // Some messages my already have string contents at this point,
+            // so stringifying them again will add inaccurate overhead.
+            const stringContents = typeof message.contents === "string" ?
+                message.contents :
+                JSON.stringify(message.contents);
+            this.messageSize[message.clientSequenceNumber] = stringContents.length;
```

Comment:
usage of clientSequenceNumber is wrong here - there will be collisions (system ops do not have it, ops from multiple clients). You want to use sequenceNumber.

Reply:
thank you, I had my suspicions.

Reply:
BTW, in salutations like that using queue is more intuitive, easier to follow code and better for perf (assuming that queue has reasonable implementation :)).
Can you please look at leveraging Deque?